spill/board.c

```diff
@@ -17,94 +17,98 @@
 
 #include "snake.h"
 #include "board.h"
+#include "constants.h"
 #include <stddef.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
+#include <stdint.h>
 
-int board[BOARD_HEIGHT][BOARD_WIDTH];
-
+long int board[BOARD_HEIGHT][BOARD_WIDTH];
+int count_food;
 
 void init_board(void) {
 	// Initialize the board with only 0s (clear blocks)
 	for (int i = 0; i < BOARD_HEIGHT; i++) {
 		for (int j = 0; j < BOARD_WIDTH; j++) {
-			board[i][j] = 0;
+			board[i][j] = BLOCK_BLANK;
 		}
 	}
 }
 
-int get_square_value(int x, int y) {
+long int get_square_value(int x, int y) {
     return board[y][x];
 }
 
-void set_square_value(int x, int y, int value) {
+void set_square_value(int x, int y, long int value) {
     board[y][x] = value;
 }
 
 // To be used under initialization, put a whole snake on the board
 void add_snake_to_board(struct Body *snake_head){
     struct Body *snake_part = snake_head;
     do {
-        set_square_value(snake_part->x, snake_part->y, BLOCK_SNAKE);
+        long int adr = (long long) snake_part;
```

Comment:
storing a long long in a long is not a good idea. What you're actually looking for is somethling like usize, which I dont think the C standard has.